File: .claude/hooks/message-display-redact-secrets.mjs
#!/usr/bin/env node
// Caviarde les secrets dans le contenu affiché (MessageDisplay)
import { readFileSync } from 'fs';
import { fileURLToPath } from 'url';

export function run(input) {
  const delta = input.delta ?? '';

  const redacted = delta
    .replace(/sk-(?:ant-api03-|proj-)[A-Za-z0-9_-]{20,}/g, '[REDACTED-ANTHROPIC-KEY]')
    .replace(/sk-[A-Za-z0-9]{20,}/g, '[REDACTED-API-KEY]')
    .replace(/ghp_[A-Za-z0-9]{36}/g, '[REDACTED-GH-TOKEN]')
    .replace(/ghs_[A-Za-z0-9]{36}/g, '[REDACTED-GH-TOKEN]')
    .replace(/Bearer [A-Za-z0-9_\-.]{20,}/g, 'Bearer [REDACTED]');

  if (redacted === delta) return null;
  return {
    hookSpecificOutput: {
      hookEventName: 'MessageDisplay',
      displayContent: redacted,
    },
  };
}

/* v8 ignore next 5 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const input = JSON.parse(readFileSync(0, 'utf8'));
  const result = run(input);
  if (result) process.stdout.write(JSON.stringify(result));
}

File: .claude/hooks/message-display-redact-secrets.py
#!/usr/bin/env python3
# @hookstack message-display-redact-secrets
"""Redacts secrets in the displayed content (MessageDisplay)."""
import json
import re
import sys

_ANTHROPIC = re.compile(r"sk-(?:ant-api03-|proj-)[A-Za-z0-9_-]{20,}")
_API_KEY = re.compile(r"sk-[A-Za-z0-9]{20,}")
_GH_PAT = re.compile(r"ghp_[A-Za-z0-9]{36}")
_GH_S = re.compile(r"ghs_[A-Za-z0-9]{36}")
_BEARER = re.compile(r"Bearer [A-Za-z0-9_\-. ]{20,}")


def run(input_data):
    delta = input_data.get("delta") or ""

    redacted = _ANTHROPIC.sub("[REDACTED-ANTHROPIC-KEY]", delta)
    redacted = _API_KEY.sub("[REDACTED-API-KEY]", redacted)
    redacted = _GH_PAT.sub("[REDACTED-GH-TOKEN]", redacted)
    redacted = _GH_S.sub("[REDACTED-GH-TOKEN]", redacted)
    redacted = _BEARER.sub("Bearer [REDACTED]", redacted)

    if redacted == delta:
        return None
    return {
        "hookSpecificOutput": {
            "hookEventName": "MessageDisplay",
            "displayContent": redacted,
        },
    }


if __name__ == "__main__":
    data = json.load(sys.stdin)
    result = run(data)
    if result:
        sys.stdout.write(json.dumps(result))
